src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
with this type you could create an implementations like this ```MachineProducer<Cat>``` etc, lets prevent it and put a bound

Reply:
done